src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,373 @@
+                        invocationOperation,
+                        removeCalleeDeclarationNode: true,
+                        cancellationToken));
+
+            return ImmutableArray.Create<CodeAction>(codeActionKeepsCallee, codeActionRemovesCallee);
+        }
+
+        private async Task<Solution> InlineMethodAsync(
+            Document document,
+            TInvocationSyntax calleeMethodInvocationNode,
+            IMethodSymbol calleeMethodSymbol,
+            TMethodDeclarationSyntax calleeMethodNode,
+            TExpressionSyntax inlineExpression,
+            SyntaxNode callerDeclarationNode,
+            ISymbol callerSymbol,
+            bool containsAwaitExpression,
+            TStatementSyntax statementContainsCallee,
+            IInvocationOperation invocationOperation,
+            bool removeCalleeDeclarationNode,
+            CancellationToken cancellationToken)
+        {
+            var methodParametersInfo = await MethodParametersInfo.GetMethodParametersInfoAsync(_syntaxFacts, document, invocationOperation, cancellationToken).ConfigureAwait(false);
+            var inlineContext = await GetInlineMethodContextAsync(
+                document,
+                calleeMethodInvocationNode,
+                calleeMethodSymbol,
+                inlineExpression,
+                statementContainsCallee,
+                methodParametersInfo,
+                cancellationToken).ConfigureAwait(false);
+            return await ChangeSolutionAsync(
+                document,
+                calleeMethodNode,
+                callerDeclarationNode,
+                callerSymbol,
+                containsAwaitExpression,
+                inlineContext,
+                removeCalleeDeclarationNode,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<Solution> ChangeSolutionAsync(Document document,
+            TMethodDeclarationSyntax calleeMethodNode,
+            SyntaxNode callerDeclarationNode,
+            ISymbol callerSymbol,
+            bool containsAwaitExpression,
+            InlineMethodContext inlineMethodContext,
+            bool removeCalleeDeclarationNode,
+            CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var documentEditor = await solutionEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
+            var statementContainsCalleeInvocationExpression = inlineMethodContext.StatementContainingCallee;
+            foreach (var statement in inlineMethodContext.StatementsToInsertBeforeCallee)
+            {
+                documentEditor.InsertBefore(
+                    statementContainsCalleeInvocationExpression,
+                    // Make sure the statement is aligned with the existing statement
+                    statement.WithTriviaFrom(statementContainsCalleeInvocationExpression));
```

Comment:
this is not a good idea unless you filter this just to leading whitespace.  you will end up with things like comment/pp-trivia duplicated.  please test.

Reply:
I think I might post this in another comments. What's I want is just aligning, is there any good service that can get that?

Reply:
Remove this part.. Suprising no test fails.